### ISSUE TITLE
Missing vector tiles on larger scales

### ISSUE THREAD

**Issue by SupaplexOSM** [gislars/strassenraum-berlin#28]
Die Kartenansicht ist bei größeren Maßstäben offenbar nur in der Lage, einen Teil der Segmente zu rendern und lässt dann "Kacheln" aus. Gibt es dafür Lösungsmöglichkeiten, außer die Segmente je nach Zoomstufe künstlich anzupassen/zu vereinfachen? Wobei ja klar ist, dass nicht beliebig viele Segmente dargestellt werden können, wenn man immer weiter herauszoomt...

![kachelfehler](https://user-images.githubusercontent.com/66696066/175787780-3251b734-6f90-4d97-a5c4-b7aed3e894a7.jpg)

**Comment by tordans** [comment]
Ja, das Problem entsteht, wenn der Tile zu groß wird, weil zu viele Daten drin sind. Daten sind sowohl die Geo-Daten als auch die Attribute.

> Gibt es dafür Lösungsmöglichkeiten, außer die Segmente je nach Zoomstufe künstlich anzupassen/zu vereinfachen? 

- Attribute reduzieren; ggf. abhängig von der Zoomstufe
- Geodaten vereinfachen (weniger geometrische Details)
- Geodaten zusammenführen (kleine Schnipsel "ignorieren" aber mit anzeigen als Teil von etwas größerem)
- Geodaten löschen (bspw. bei parallelen Linien nur noch eine anzeigen)

Ich habe in der Vergangenheit erfolgreich mit https://github.com/mapbox/tippecanoe gearbeitet (das aber hier technisch wahrscheinlich nicht die richtige Wahl ist). Dort kann man Schwellenwerte definieren und angeben, ob beim Vereinfachen dann Segmente wegfallen sollen oder zusammengeführt werden sollen. Wenn sie zusammengeführt werden, bekommt das neue Element einen count, der darauf hinweist.

Ein andere Plan ist, für unterschiedliche Zoom-Stufen unterschiedliche Berechnungen / Daten zu verwenden und diese dann in einen Vektor-Datensatz zusammen zuführen.

Was für unseren "tech stack" die richtige Lösung ist, weiß vor allem Lars am besten.

Ich würde das Thema aber vertagen, bis wir weiter sind.

**Idee**

- [ ] Ich könnte bis dahin das raus-zoomen verhindern in der Karte. Dann ist es zumindest nicht  defekt …

**Comment by SupaplexOSM** [comment]
Das Raus-zoomen verhindern könnte nervig sein, da das Problem zur Zeit recht früh entsteht und man dann nicht wirklich schön über die Stadt fliegen kann. Dann würde ich lieber erstmal damit leben bis wir später überlegen, welche "Aussage" die Daten in diesen Zoomstufen transportieren sollten und sie entsprechend optimieren.

**Comment by gislars** [comment]
Das Verhalten ist mir auch aufgefallen, aber ich hatte noch keine Erklärung dafür. Zu viele Daten in einem Tile... klingt plausibel. Das Thema würde ich auch erstmal vertagen. Es gibt einige Möglichkeiten das Datenvolumen zu reduzieren, ich habe in der Richtung hier noch nichts gemacht.

raus-zoomen verhindern fände ich auch nicht gut. So lassen wie es ist, bis wir mehr an der Visualisierung (Tiles, Kartenstil etc) arbeiten
